app/controllers/katello/api/registry/registry_proxies_controller.rb

```diff
@@ -105,7 +112,8 @@ def find_readable_repository
     end
 
     def require_user_authorization?(repository = @repository)
-      !repository || (!repository.environment.registry_unauthenticated_pull && !ssl_client_authorized?(repository.organization.label))
+      !(params['action'] == 'token' && params['scope'].blank? && params['account'].blank?) &&
```

Comment:
what was the reason to add this line?

Reply:
It was to allow for unscoped requests to the `token` endpoint, so that unauthenticated requests could proceed to the `_catalog` method.

Reply:
The original version specifically prohibited unscoped token requests.